app/controllers/create-service.controller.js

```diff
@@ -38,15 +38,19 @@ exports.post = (req, res) => {
       current_name: serviceName,
       current_name_cy: serviceNameCy
     })
-    res.redirect(paths.serviceSwitcher.create)
-  } else {
-    return serviceService.createService(serviceName, serviceNameCy, correlationId)
-      .then((service) => userService.assignServiceRole(req.user.externalId, service.external_id, 'admin', correlationId))
-      .then(() => {
-        res.redirect(paths.serviceSwitcher.index)
-      })
-      .catch(err => {
-        renderErrorView(req, res, err)
-      })
+    return res.redirect(paths.serviceSwitcher.create)
+  }
+
+  try {
+    const service = await serviceService.createService(serviceName, serviceNameCy, req.user, correlationId)
+    await userService.assignServiceRole(req.user.externalId, service.externalId, 'admin', correlationId)
+    res.redirect(paths.serviceSwitcher.index)
+  } catch (err) {
+    renderErrorView(req, res, err)
```

Comment:
I believe passing errors through to `next` will do the equivalent of a default render error view, that would be worth double checking though!
```suggestion
    next(err)
```

Reply:
Yes it does, I almost did this but then was worried about having to update tests. I can do it

Reply:
Done